core/swupdate.c

```diff
@@ -772,9 +772,14 @@ int main(int argc, char **argv)
 			break;
 #ifdef CONFIG_DOWNLOAD
 		case 'd':
-			(void)snprintf(dwloptions, sizeof(dwloptions), "%s %s", argv[0], optarg);
+			if (asprintf(&dwloptions,"%s %s", argv[0], optarg) ==
+			ENOMEM_ASPRINTF) {
+				ERROR("Error in user input. Sufficient amount of memory cannot be allocated.");
+				break;
```

Comment:
@TomaszAIR Have you checked the application's behavior when code executions enters here?
For me, it just hangs execution and CTR+C must be used to exit from the application. It should rather exit with error message, without additional SIGINT.